File: L2000-reverse_prefix_of_word.py
# https://leetcode.com/problems/reverse-prefix-of-word/
class Solution:
    def getIndex(self, word: str, ch: str):
        try:
            return word.index(ch)
        except:
            return -1
    
    def reversePrefix(self, word: str, ch: str) -> str:
        idx = self.getIndex(word, ch)
        length = len(word)
        res = ""
        for i in range(idx, -1, -1):
            res += word[i]
        for i in range(idx + 1, length):
            res += word[i]
        return res

File: L2000-ReversePrefixOfWord.java
// https://leetcode.com/problems/reverse-prefix-of-word/
class Solution {
    public String reversePrefix(String word, char ch) {
        int idx = word.indexOf(ch);
        if (idx == -1) {
            return word;
        }
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < word.length(); i++) {
            if (i <= idx) {
                result.append(word.charAt(idx - i));
            }
            else {
                result.append(word.charAt(i));
            }
        }
        return result.toString();
    }
}
